### PR TITLE
Feature 3 building houses and hotels

### PR DESCRIPTION
**Pull request by SDunk1011**

### CODE REVIEW

File: src/main/java/TurnUseCases/BuildUseCase/BuildBuildingInputBoundary.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package TurnUseCases.BuildUseCase;
+
+import GameEntities.Player;
+import GameEntities.Tiles.ColorPropertyTile;
+import GameEntities.Tiles.Property;
+
+public interface BuildBuildingInputBoundary {
```
Comment (JLenander) [inline src/main/java/TurnUseCases/BuildUseCase/BuildBuildingInputBoundary.java:7]: Do we want BuildBuilding use case to also have methods for selling houses and hotels?

File: src/main/java/TurnUseCases/BuildUseCase/BuildBuildings.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+     */
+    @Override
+    public boolean isBuildable (Player player, Property property){
+        if(!player.ownsProperty(property)){
+            return false;
+        }
+        if(property.isMortgaged()){
+            return false;
+        }
+        if(!property.getClass().getName().equals("GameEntities.Tiles.ColorPropertyTile")){
```
Comment (JLenander) [inline src/main/java/TurnUseCases/BuildUseCase/BuildBuildings.java:31]: If you're checking the class of the object, you can use `property instanceof ColorPropertyTile`

**Review by JLenander — CHANGES_REQUESTED** [review]
Lines 34-62 in `BuildBuildings.java` isn't necessary and should use the entity method. The ColorPropertyTile class has a method for checking if they own the whole set (`colorPropertyTile.checkSetOwned`).
Lines 86-104 in `BuildBuildings.java` should use the method `colorPropertyTile.getBuildingCost` (same for the `buildHotel` method.
See e40f2aae15067f77e9d9bc59dfa160ec495e37f5 for the updated ColorPropertyTile code

**Comment by JLenander** [comment]
Same as #34 , the Presenter being in all of the method signatures indicates that you can have the presenter as an instance variable (passing it into the constructor) instead of passing it into every method.

File: src/main/java/TurnUseCases/BuildUseCase/BuildBuildings.java [hunk 3]
```diff
@@ -81,7 +98,8 @@ public boolean buildHouse (Player player, ColorPropertyTile property){
      */
     public boolean buildHotel (Player player, ColorPropertyTile property){
         if (isBuildable(player, property) && property.getHouses() >= 4){
-            property.hotels += 1;
+            property.addHotel();
+            //need to add a method addHotel in ColorPropertyTile class to add the number of hotel.
```
Comment (Youssefb1234) [inline src/main/java/TurnUseCases/BuildUseCase/BuildBuildings.java:105]: I added the add house and add hotel methods in color property tile so you should be able to implement the remaining build methods.

**Review by alirezazim03 — APPROVED** [review]

**Review by iamyanbo — APPROVED** [review]

**Review by JLenander — APPROVED** [review]

**Review by TahirLeonenko — APPROVED** [review]

**Pull request merged**